packages/ssx-sdk/src/modules/Storage.ts

```diff
@@ -1,42 +1,278 @@
+   * @returns A Promise that resolves when the operation is complete.
+   */
+  unencrypted_put(key: string, value: any): Promise<void>;
+
+  /**
+   * Retrieves the encrypted stored value associated with the specified key, decrypting it before returning.
+   * @param key - The unique identifier for the stored value.
+   * @returns A Promise that resolves to the decrypted value associated with the given key or undefined if the key does not exist.
+   */
+  get(key: string): Promise<any>;
+
+  /**
+   * Stores a value with the specified key, encrypting it before storage.
+   * @param key - The unique identifier for the stored value.
+   * @param value - The value to store under the given key with encryption.
+   * @returns A Promise that resolves when the operation is complete.
+   */
+  put(key: string, value: any): Promise<void>;
 }
 
-// class BrowserStorage implements IStorage {}
-// class BrowserDataVault extends BrowserStorage implements IDataVault {
-class BrowserDataVault implements IDataVault {
-  private encryption: IEncryption;
+/**
+ * Represents a storage configuration object with an optional prefix.
+ */
+interface IStorageConfig {
+  prefix?: string;
+}
 
-  constructor(config: any, encryption: IEncryption) {
-    this.encryption = encryption;
+/**
+ * Represents a browser storage configuration object that extends IStorageConfig.
+ */
+interface IBrowserStorageConfig extends IStorageConfig {
+  storeName?: string;
+}
+
+/**
+ * Represents a class for managing browser storage operations.
+ */
+class BrowserStorage implements IStorage {
+  private prefix: string;
+  private dbName: string;
+  private storeName: string;
+
+  /**
+   * Constructs a new instance of BrowserStorage.
+   * @param config - An object containing optional configuration properties for browser storage.
+   */
+  constructor(config: IBrowserStorageConfig) {
+    // if no window object, throw error
+    if (typeof window === 'undefined') {
+      throw new Error('BrowserStorage: window object not found');
+    }
+
+    this.prefix = config?.prefix || '';
+    this.dbName = 'ssx-browser-storage';
+    this.storeName = config?.storeName || 'ssx-browser-storage-store';
+  }
+
+  /**
+  * Prefixes the provided key based on the stored prefix.
+  * @param key - The key to be prefixed.
+  * @returns The prefixed key.
+  */
+  private prefixKey(key: string): string {
+    return this.prefix ? `${this.prefix}/${key}` : key;
+  }
+
+  /**
+   * Retrieves the underlying IDBPDatabase instance.
+   * @returns A Promise that resolves to the IDBPDatabase instance.
+   */
+  private async getDB(): Promise<IDBPDatabase> {
+    return await openDB(this.dbName, 1, {
+      upgrade(db) {
+        if (!db.objectStoreNames.contains(this.storeName)) {
+          db.createObjectStore(this.storeName);
+        }
+      },
+    });
+  }
+
+  public async get(key: string): Promise<any> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readonly');
```

Comment:
maybe renaming `tx` to `transaction` will look cleaner

Reply:
Done